File: subsidy-platform/src/main/java/com/subsidy_platform/service/SchemeService.java
package com.subsidy_platform.service;

import com.subsidy_platform.dto.SchemeRequest;
import com.subsidy_platform.dto.SchemeResponse;
import java.util.List;

public interface SchemeService {
    List<SchemeResponse> getAll();
    SchemeResponse getById(Long id);
    SchemeResponse create(SchemeRequest request);
    SchemeResponse update(Long id, SchemeRequest request);
}

File: frontend/src/services/schemeService.ts
import apiClient from './apiClient';
import type { SchemeRequest, SchemeResponse } from '../types/api';

export const schemeService = {
  getAll: async (): Promise<SchemeResponse[]> => {
    const response = await apiClient.get<SchemeResponse[]>('/schemes');
    return response.data;
  },

  getById: async (id: number): Promise<SchemeResponse> => {
    const response = await apiClient.get<SchemeResponse>(`/schemes/${id}`);
    return response.data;
  },

  create: async (data: SchemeRequest): Promise<SchemeResponse> => {
    const response = await apiClient.post<SchemeResponse>('/schemes', data);
    return response.data;
  },

  update: async (id: number, data: SchemeRequest): Promise<SchemeResponse> => {
    const response = await apiClient.put<SchemeResponse>(`/schemes/${id}`, data);
    return response.data;
  },
};

export default schemeService;
